.github/workflows/build.yml

```diff
@@ -0,0 +1,83 @@
+name: Build and Deploy
+
+on:
+  push:
+    branches:
+      - staging
+      - master
+  pull_request:
+    branches:
+      - staging
+      - master
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Checks, Tests, & Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v4
+
+      - name: Node.js setup
+        uses: actions/setup-node@v4
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: ✅ Run Tests
+        run: yarn test --no-watch --coverage
+
+      - name: 🏗️ Build app
+        run: yarn build
```

Comment:
Does this build not need `REACT_APP_GLOBAL_REGISTRY_BASE_URL: ${{ secrets.GLOBAL_REGISTRY_BASE_URL }}`?

Reply:
Not really because the output is discarded. It's more of a CI check to make sure that the app builds correctly. But maybe I should add it anyway.